modules/margin/src/main/java/com/opengamma/sdk/margin/MarginClient.java

```diff
@@ -132,4 +221,25 @@ public abstract MarginWhatIfCalcResult calculateWhatIf(
       MarginCalcRequest request,
       List<PortfolioDataFile> deltaFiles);
 
+  /**
+   * High-level call to submit a base portfolio together with an extra set of trades,
+   * for parsing, validation and IM calculation.
+   * <p>
+   * This will return the margin summary for the base request, the combined request (base portfolio + delta portfolio),
+   * and the difference between the two.
+   *
+   * @param ccp  the CCP to use
+   * @param request  the calculation request
+   * @param deltaFiles  the portfolios representing the extra trades for the what-if scenario
+   * @param retries the number of times to retry a call that failed due to IO or network errors
+   * @return the detailed result of the calculation
+   * @throws MarginException if unable to calculate
+   * @throws UncheckedIOException if an IO error occurs
+   */
+  public abstract MarginWhatIfCalcResult calculateWhatIf(
+      Ccp ccp,
+      MarginCalcRequest request,
+      List<PortfolioDataFile> deltaFiles,
+      int retries);
```

Comment:
I don't think this should be an argument to the method. It should be part of the setup of the service client